src/algebra/module/polynomial.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2022 Pierre-Alexandre Bazin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pierre-Alexandre Bazin
+-/
+import data.polynomial
+
+/-!
+# Modules over `R[X]`
+
+## Main definitions
+
+* `module_with_End E` : A type synonym for a `R`-module which explicitly mentions a given
+  endomorphism `E`. It naturally comes with a `R[X]`-module structure, where the action of `X` is
+  given by `E`.
+
+## Main statements
+
+* `module_with_End.polynomial_linear_equiv_semiconj` : equivalence between `R[X]`-linear maps and
+  `R`-linear maps compatible with the action of `X`. See also
+  `module_with_End.polynomial_lequiv_equiv_semiconj` for the restriction to linear isomorphisms.
+
+## Tags
+
+Module, polynomial
+-/
+
+
+variables {R M : Type*} [comm_semiring R] [add_comm_monoid M] [module R M]
+open_locale polynomial
+
+
+open polynomial module
+
+/--A type synonym for a `R`-module which explicitly mentions a given endomorphism.-/
+@[nolint unused_arguments has_inhabited_instance]
+def module_with_End (E : End R M) := M
+
+namespace module_with_End
+variables {E : End R M}
+
+instance : add_comm_monoid (module_with_End E) := (infer_instance : add_comm_monoid M)
```

Comment:
If you change line 36 to `@[nolint unused_arguments has_inhabited_instance, derive add_comm_monoid]` then you don't need line 42.